### PR TITLE
Add initialStatus

### LINKED ISSUE #471: Initialization of status
## Bug, Feature, or Question?
Bug

## Current Behavior
When including a reference to `this.status.property` in a Formik inner component, where `property` is determined after `handleSubmit` is called, initially referencing `this.status.property` results in error: Cannot find property 'property' of undefined

## Desired Behavior
Reference to `this.status.property` should return `undefined` on component initialization.

## Suggested Solutions
After a few hours of digging, and looking at [this](https://codesandbox.io/s/ppnkqv78wq) CodeSandbox, the current solution is to append `this.status &&` before `this.status.property` so the conjunction short-circuits to false.

Given this is a basic JS fact of how objects evaluate, it may seem like the solution is obvious, but I found the documentation regarding how `status` is handled a little sparse. 

Maybe there could be a `initializeStatus` property for the Formik component which allows you to specify `status` on page initialization?

- Formik Version: 0.11.11
- React Version: 16.0
- TypeScript Version: N/A
- CodeSandbox Link: https://codesandbox.io/s/ppnkqv78wq
- OS: Windows 10
- Node Version: 9.5
- Package Manager and Version: npm 5.6.0

### PR DESCRIPTION
**Pull request by charlax**
Fixes #471 (proof of concept, lacking tests)

### CODE REVIEW

**Comment by charlax** [comment]
@jaredpalmer what do you think about this?

**Comment by jaredpalmer** [comment]
I think it initialStatus makes sense. However, I think we need to see some real examples though of exactly what this solves for and then decide if it is still necessary against alternatives.

**Comment by jaredpalmer** [comment]
I reread 471 and I’m still unconvinced that status preferred in these situations vs. react state. 

Cc @eonwhite @slightlytyler have you run into the need for this? Recall that status replaced the singular error and setError which were initially meant for a form level error message. It initialized to undefined because we don’t use null at The Palmer Group. It then became a kind of store whatever you want escape hatch. As hooks arrive, I think that status might be a bit of an anti-pattern. Perhaps the only lingering value being that status gets reset with the rest of the form in resetForm(), which I guess is legit, but idk.

**Comment by charlax** [comment]
You're right that this is kind of using `status` for something else than what was originally intended. I could achieve the same thing with building my own context instead of putting this in `status`.

That being said, as soon as you provide `status`, I think you need to allow setting `initialStatus`, just like Downshift provides `initialX` for all of its `X` state. And you provide `initialValues` for `values` so...

**Comment by jaredpalmer** [comment]
I agree. This would also solve for the monstrosity of isInitialInvalid and isValid. However, then we should not stop there and we ought to bring back up the controlled Formik RFC #812 and #837

**Comment by eonwhite** [comment]
I barely use status honestly -- but I agree that as long as it exists, it makes sense to have an option to populate its initial value, just for consistency of API.

File: src/types.tsx [hunk 1]
```diff
@@ -191,6 +191,11 @@ export interface FormikConfig<Values> extends FormikSharedConfig {
    */
   initialValues: Values;
 
+  /**
+   * Initial status
+   */
+  initialStatus: any;
```
Comment (jaredpalmer) [inline src/types.tsx]: Make this optional.
```suggestion
  initialStatus?: any;
```

**Comment by jaredpalmer** [comment]
I'm going to finish this all up this afternoon

**Comment by jaredpalmer** [comment]
See #1321

**Comment by charlax** [comment]
Thanks @jaredpalmer !

**Pull request merged**